plugins/inputs/cloudwatch/cloudwatch.go

```diff
@@ -208,14 +215,32 @@ func (c *CloudWatch) Gather(acc telegraf.Accumulator) error {
 		<-lmtr.C
 		go func(inm *cloudwatch.Metric) {
 			defer wg.Done()
-			acc.AddError(c.gatherMetric(acc, inm, now))
+			acc.AddError(c.gatherMetric(acc, inm))
 		}(m)
 	}
 	wg.Wait()
 
 	return nil
 }
 
+func (c *CloudWatch) updateWindow(relativeTo time.Time) error {
+	defaultTime := time.Time{}
+
+	windowEnd := relativeTo.Add(-c.Delay.Duration)
+
+	if c.windowEnd == defaultTime {
```

Comment:
Nitpick: `if c.windowEnd.IsZero()`

Reply:
Didn't know that function was there. I'll update

As for the rounding, is there something you had in mind? It looks like, based on [this](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#CloudWatchPeriods) that non-custom metrics have a minimum period of 60s. It also says:

>When statistics are aggregated over a period of time, they are stamped with the time corresponding to the beginning of the period.

Seems to indicate it should be aligned to the start of the minute. Is this what you're referring to?

Reply:
On the rounding, it's probably not something that needs done as I don't see us doing it before.  I was just curious if the times needed to rounded and aligned, but I guess either of these ranges would work fine (second one is not aligned to period) and what really matters is that we cover all times?
```
--start-time 2018-09-06T17:15:00Z --end-time 2018-09-06T17:30:00Z --period 300
--start-time 2018-09-06T17:16:42Z --end-time 2018-09-06T17:31:42Z --period 300
```

Reply:
I fixed the zero function.

I think we probably don't need to align the data as long as we cover all time periods, like you said.